### PR TITLE
Add redis integration

### PR DESCRIPTION
**Pull request by dafydd-t**
Delivers: https://github.com/grafana/integrations-pipeline/issues/12

Verified at container level by running the docker-compose in `example/docker-compose.integrations.yaml` with the following added into `agent.yaml`:
```yaml
integrations:
  redis_exporter:
    enabled: true
    redis_addr: 'redis:6379'
```

### CODE REVIEW

**Comment by rfratto** [comment]
Is this ready for review @dafydd-t?

File: go.mod
Comment (dafydd-t) [inline go.mod:27]: This bump is because redis_exporter has 1.7.0 as a dependency

File: pkg/integrations/redis_exporter/redis_exporter.go
Comment (dafydd-t) [inline pkg/integrations/redis_exporter/redis_exporter.go:42]: I wanted to be consistent with mysql here: one of `redis_addr` or `REDIS_EXPORTER_ADDRESS` must be set for the integration to work.

**Comment by dafydd-t** [comment]
> Is this ready for review @dafydd-t?

@rfratto it is now!

File: go.mod
Comment (rfratto) [inline go.mod:73]: I think you could probably omit a lot of the detail here.

```suggestion
// Required for redis_exporter, which is incompatible with v2.0.0+incompatible.
```

File: pkg/integrations/redis_exporter/redis_exporter.go
Comment (rfratto) [inline pkg/integrations/redis_exporter/redis_exporter.go:49]: Use the standard go error wrapping here: 

```suggestion
			return nil, fmt.Errorf("Error loading script file %s: %w", c.ScriptPath, err)
```

File: pkg/integrations/redis_exporter/redis_exporter.go
Comment (rfratto) [inline pkg/integrations/redis_exporter/redis_exporter.go:62]:
```suggestion
			return nil, fmt.Errorf("couldn't load TLS client key pair: %w", err)
```

File: pkg/integrations/redis_exporter/redis_exporter_test.go
Comment (rfratto) [inline pkg/integrations/redis_exporter/redis_exporter_test.go:28]: Small nit: test tables should be defined inline in the test that uses them with the following pattern:

```
tt := []struct{
  name string 
  cfg Config 
  expectedMetrics []string 
  expectedConstructorError bool 
  envVars map[string]string 
}{
  // ... 
}

// test case in test table 
for _, tc := range tt {
  t.Run(tc.name, func(t *testing.T) {
     // run test case 
  })
}
```

**Review by rfratto — COMMENTED** [review]
Nice, this is looking really good! 

Can you put the temporary fork of redis_exporter under grafana and put your changes there instead so we all have control over it?

File: docs/configuration-reference.md
Comment (hoenn) [inline docs/configuration-reference.md:2197]:
```suggestion
The `redis_exporter_config` block configures the `redis_exporter` integration, which is an embedded version of [`redis_exporter`](https://github.com/oliver006/redis_exporter). This allows for the collection of metrics from Redis servers.
```

File: docs/configuration-reference.md
Comment (dafydd-t) [reply docs/configuration-reference.md:2255]: I did not think of this before your comment, but redis also has this [URI format](https://github.com/lettuce-io/lettuce-core/wiki/Redis-URI-and-connection-details#uri-syntax) that allows credentials to be provided, e.g. `redis://user:pwd@redis:6379`. Do you think this makes it more useful to keep the `REDIS_EXPORTER_ADDRESS` envVar option?

aside from that, I'll add a `redis_password_file` config property that takes precedence over `redis_password`

Comment (dafydd-t) [reply docs/configuration-reference.md:2255]: Yes. I I thought I was being consistent with the redis_exporter property names, but I'm not here!

Comment (rfratto) [reply docs/configuration-reference.md:2255]: Nah, let’s not even document the URI format and only encourage the one way we’ve already defined. As long as they have a way to provide a password without putting it directly in the config file, I don’t think we need an environment variable.

File: pkg/integrations/redis_exporter/redis_exporter_test.go
Comment (dafydd-t) [reply pkg/integrations/redis_exporter/redis_exporter_test.go:28]: nice, I assume this avoids test runs being cached, even though we've changed the test input.

File: docs/configuration-reference.md
Comment (dafydd-t) [reply docs/configuration-reference.md]: OK I'll chuck it.

File: pkg/integrations/redis_exporter/redis_exporter_test.go
Comment (rfratto) [inline pkg/integrations/redis_exporter/redis_exporter_test.go:106]: Can you use [subtests](https://golang.org/pkg/testing/#T.Run) here? 

```go
for _, test := range tt {
  t.Run(test.name, func(t *testing.T) {
     integration, err := New(logger, test.cfg)
     if test.expectConstructorError {
         require.Error(t, err, "expected error when setting up exporter") 
         return
     } 
     require.NoError(t, err, "failed to setup exporter")
     
     // ... etc ...
  })
}
```

**Review by rfratto — APPROVED** [review]
LGTM! I have one final nit but you don't need a second approval after it's fixed.

After you're done, feel free to merge this (just make sure you do a squash and merge and not a squash/merge commit). Nice work!

File: pkg/integrations/redis_exporter/redis_exporter_test.go
Comment (rfratto) [inline pkg/integrations/redis_exporter/redis_exporter_test.go:108]: Forgot to mention, but you can take out the env vars bit now :)

File: pkg/integrations/redis_exporter/redis_exporter_test.go
Comment (rfratto) [reply pkg/integrations/redis_exporter/redis_exporter_test.go]: Nah, there's no functional difference in doing it one way over the other (but subtests can be filtered out at runtime to focus on a specific test case). 

The rest of the codebase defines the test cases inside the test function, so I just want to be consistent with that.

File: pkg/integrations/redis_exporter/redis_exporter_test.go
Comment (dafydd-t) [reply pkg/integrations/redis_exporter/redis_exporter_test.go:106]: Nice, yes - we get the test case names for free then.

**Pull request merged**
